presto-main/src/main/java/com/facebook/presto/operator/TopNOperator.java

```diff
@@ -162,29 +160,33 @@ public Page getOutput()
             return null;
         }
 
-        if (outputIterator == null) {
+        if (outputPages == null) {
             // start flushing
-            outputIterator = topNBuilder.buildResult();
+            outputPages = topNBuilder.buildResult();
         }
 
-        Page output = null;
-        if (outputIterator.hasNext()) {
-            output = outputIterator.next();
+        if (outputPages == null) {
+            // start flushing
+            outputPages = topNBuilder.buildResult();
         }
```

Comment:
Duplicated code?

Reply:
Yes I previously attempted to squash commits and looks like this is an artifact from that. Will clean up